### DISCUSSION TITLE
how to enable this tcp fast open and VMESS AEAD?

### DISCUSSION THREAD

**Discussion by malikshi** [XTLS/Xray-core#388] [Q&A] [answered]
hello, where config do i need to changes to enable the feature TCP Fast Open and Vmess AEAD?
>TCP Fast Open
Linux Server supports specifying the TCP Fast Open queue length
"tcpFastOpen": true The default queue length has been changed from 1 to 256
VMess
Add VMess zero encryption (but still slower than VLESS and Trojan) 
You can set XRAY_VMESS_AEAD_FORCED = true at server side

**Comment by ghost** [comment, marked as answer]
Vmess AEAD is enabled if `alterID` = 0.
You can enable TFO in [sockopt](https://xtls.github.io/Xray-docs-next/config/transport.html#sockoptobject).

**Reply by malikshi** [reply]
thank you @JimhHan  
I set it like this.
> "streamSettings": {
    "network": "ws",
    "security": "auto",
    "wsSettings": {
      "path": "/path"
    },
    "sockopt": {
      "mark": 0,
      "tcpFastOpen": true
    }
  }

with AlterID zero, it is much better in ram usage.
